src/legacy/server/config/transform_deprecations.test.js

```diff
@@ -62,5 +61,109 @@ describe('server/config', function () {
         sinon.assert.calledWithExactly(log, sinon.match('savedObjects.indexCheckTimeout'));
       });
     });
+
+    describe('csp.rules', () => {
+      describe('with nonce source', () => {
+        it('logs a warning', () => {
+          const settings = {
+            csp: {
+              rules: [`script-src 'self' 'nonce-{nonce}'`],
+            },
+          };
+
+          const log = jest.fn();
+          transformDeprecations(settings, log);
+          expect(log.mock.calls).toMatchInlineSnapshot(`
+            Array [
+              Array [
+                "csp.rules no longer supports the {nonce} syntax. Replacing with 'self' in script-src",
+              ],
+            ]
+          `);
+        });
+
+        it('removes a quoted nonce', () => {
```

Comment:
For the tests that remove the nonce, can we not include `'self'` in the initial rules and ensure we add 'self' here (even if they don't match script-src) or add dedicated tests for this?

Reply:
Done!